src/controller/user-controller.ts

```diff
@@ -876,6 +884,42 @@ export default class UserController extends BaseController {
     }
   }
 
+  /**
+   * GET /users/nfc/{id}
+   * @summary Get a user using the nfc code
+   * @operationId findUserNfc
+   * @tags users - Operations of the user controller
+   * @security JWT
+   * @param {integer} id.path.required - The nfc code of the user
+   * @return {UserResponse} 200 - The requested user
+   * @return {string} 404 - The user with the given nfc code does not exist
+   */
+  public async findUserNfc(req: RequestWithToken, res: Response): Promise<void> {
+    const parameters = req.params;
+    this.logger.trace('Find user nfc', parameters, 'by user', req.token.user);
+
+    try {
+      const nfcCode = String(parameters.id);
+      const nfc = await NfcAuthenticator.findOne({ where: { nfcCode } });
```

Comment:
Can't you just use `{ where: { nfcCode }, relations: { user: true } }` such that the user is automatically returned? That makes the condition on line 911 obsolete (which also does not really make sense, as all NFC authenticators must have a corresponding user, but I see why you had to do it).

Reply:
Kinda, the UserService has some tech debt, so this will mean that the response will always be a UserResponse and not a GEWIS UserResponse. Fixing this is outside the scope of this PR, and is therefore a trade-off.

How important is that the GEWIS id is also returned?